File: Jardin De infantes/UI/Servicios/Login/iuLogin.vb
﻿Imports BLL
Imports EE
Imports MetroFramework

Imports System.Globalization
Imports System.Threading


Public Class iuLogin
    Dim idioma As String



    Sub cambiaIdiomaForm()

        Dim unbllTraducir As New bllTraduccion
        Dim unIdioma As New Idioma(txtIdioma.Text)
        SingletonIdioma.intance.NuevoIdioma(unIdioma)
        Dim unTraducir As New Traduccion(SingletonIdioma.intance.getUsuario)
        Dim ht As New Hashtable
        ht = unbllTraducir.ListarPalabrasTraducidas(unTraducir)
        For Each item As DictionaryEntry In ht
            For Each controlcito As Control In Me.Controls
                If controlcito.Tag Is Nothing Then
                Else
                    If controlcito.Tag.Equals(item.Key) Then
                        controlcito.Text = item.Value

                    End If
                End If





            Next
        Next



    End Sub
    Sub CambiarCultura()

        Dim culturaObj As CultureInfo = New CultureInfo(txtIdioma.Text)
        Threading.Thread.CurrentThread.CurrentUICulture = culturaObj
        Threading.Thread.CurrentThread.CurrentCulture = culturaObj
    End Sub
    Private Sub iuLogin_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim unBll As New bllIdioma
        txtIdioma.DataSource = unBll.listaIdiomaString
        cambiaIdiomaForm()
        CambiarCultura()





    End Sub

    Private Sub txtIdioma_SelectedIndexChanged(sender As Object, e As EventArgs) Handles txtIdioma.SelectedIndexChanged
        cambiaIdiomaForm()
        CambiarCultura()

    End Sub

    Private Sub btnAceptar_Click(sender As Object, e As EventArgs) Handles btnAceptar.Click


        Dim unbllusuario As New bllUsuario

        If unbllusuario.VerificarCompletoUsuario(txtPassword.Text, txtUsuario.Text, txtIdioma.Text) = True Then



            MsgBox("Bienvenido " & sessionManager.intance.getUsuario.getPersona.getNombre, , "")
            Dim frmPrincipa As New iuHOme

            frmPrincipa.ShowDialog()
            Me.Close()

        Else
            MetroMessageBox.Show(Me, "Usuario o Contraseña invalidas", "Error en el login", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End If




    End Sub

    Private Sub lblUsuario_Click(sender As Object, e As EventArgs) Handles lblUsuario.Click

    End Sub

    Private Sub btnCancelar_Click(sender As Object, e As EventArgs) Handles btnCancelar.Click
        Me.Close()
    End Sub

End Class

File: Jardin De infantes/Seguridad/IU/iuLogin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using EEseguridad;
using BLLseguridad.Login;
using BLLseguridad.Idioma;
using System.Globalization;
using System.Collections;
using IU.Properties;

namespace IU
{
    public partial class iuLogin : MetroFramework.Forms.MetroForm
    {
        public iuLogin()
        {
            InitializeComponent();
          
        }

        private void iuLogin_Load(object sender, EventArgs e)
        {
          
        }

        private void iuLogin_Load_1(object sender, EventArgs e)
        {
            helpProvider1.HelpNamespace =CHM.Cadena;
            bllIdioma bll = new bllIdioma();
            DataTable dt = new DataTable();
            dt = bll.ListarIdiomas();

            foreach (DataRow item in dt.Rows)
            {
                cbIdioma.Items.Add(item["Idioma"].ToString());
            }
            cbIdioma.SelectedIndex = 0;
            CambiarCultura();
        }
        void cambiarIdioma(Form iu)
        {
            bllTraduccion bll = new bllTraduccion();
            EEseguridad.Idioma.eeIdioma idioma = new EEseguridad.Idioma.eeIdioma(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            EEseguridad.Idioma.SingletonIdioma.intance.NuevoIdioma(idioma);


            EEseguridad.Idioma.SingletonIdioma.intance.CambiarIdioma(idioma.GetIdioma());
            bllTraduccion traduccion = new bllTraduccion();
            Hashtable ht = new Hashtable();
            ht = traduccion.ObtenerTraduccion(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            foreach (DictionaryEntry item in ht)
            {
                foreach (Control controles in iu.Controls)
                {
                    if (controles.Tag==null)
                    {

                    }
                    else
                    {
                        if (controles.Tag.Equals(item.Key))
                        {
                            controles.Text = item.Value.ToString();
                        }
                    }
                }
            }
        }
        private void btnAcepta_Click(object sender, EventArgs e)
        {
            EEseguridad.Login.eeUsuario unUsuario = new EEseguridad.Login.eeUsuario(txtUsuario.Text, txtContraseña.Text, "", true);
            bllLogin bll = new bllLogin();

            if (bll.VerificarLogin(unUsuario)==true)
            {
                 iuBienvenidad iu= new iuBienvenidad();
                cambiarIdioma(iu);

                iu.ShowDialog();

            }
            else
            {
                MessageBox.Show(this, "La contraseña o el usuario no es valido", "Ingreso", MessageBoxButtons.OK);
                txtContraseña.Text = "";

            }
        }

        private void txtContraseña_TextChanged(object sender, EventArgs e)
        {

        }
        void  CambiarCultura()
        {
            CultureInfo cultura = new CultureInfo(cbIdioma.Text);
            System.Threading.Thread.CurrentThread.CurrentUICulture = cultura;
            System.Threading.Thread.CurrentThread.CurrentCulture = cultura;

        }
        void CambiarIdioma()
        {
            bllTraduccion bll = new bllTraduccion();
            EEseguridad.Idioma.eeIdioma idioma = new EEseguridad.Idioma.eeIdioma(cbIdioma.Text);
            EEseguridad.Idioma.SingletonIdioma.intance.CambiarIdioma(idioma.GetIdioma());


            EEseguridad.Idioma.SingletonIdioma.intance.CambiarIdioma(idioma.GetIdioma());
            bllTraduccion traduccion = new bllTraduccion();
            Hashtable ht = new Hashtable();
            ht = traduccion.ObtenerTraduccion(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            foreach (DictionaryEntry item in ht)
            {
                foreach (Control controles in this.Controls)
                {
                    if (controles.Tag==null)
                    {

                    }
                    else
                    {
                        if (controles.Tag.Equals(item.Key))
                        {
                            controles.Text = item.Value.ToString();
                        }
                    }
                }
            }
                
           
        }
        private void cbIdioma_SelectedIndexChanged(object sender, EventArgs e)
        {
            CambiarCultura();
            EEseguridad.Idioma.eeIdioma idioma = new EEseguridad.Idioma.eeIdioma(cbIdioma.Text);
            EEseguridad.Idioma.SingletonIdioma.intance.NuevoIdioma(idioma);
          
            CambiarIdioma();
           
        }
    }
}
